modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/helper/SaveFormInstanceMVCCommandHelper.java

```diff
@@ -105,7 +106,24 @@ public DDMFormInstance saveFormInstance(
 			PortletRequest portletRequest, PortletResponse portletResponse)
 		throws Exception {
 
-		return saveFormInstance(portletRequest, portletResponse, false);
+		HttpServletRequest httpServletRequest = _portal.getHttpServletRequest(
+			portletRequest);
+
+		try {
+			AuthTokenUtil.checkCSRFToken(
+				httpServletRequest,
+				SaveFormInstanceMVCCommandHelper.class.getName());
+
```

Comment:
@rijema do we need to do this in [here](https://github.com/liferay-forms/liferay-portal/blob/cbfe72bb355fb12294e29b0b02f934b9d8ab11ab/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/helper/SaveFormInstanceMVCCommandHelper.java#L129-L135) too?

Reply:
If we do, I believe it is enough to check the CSRF only in the method I mentioned, since the method you modified is calling the other one.

Reply:
Hi @carolmariaabb . It does make sense as the same function pointed by you is used in other places like PublishFormInstanceMVCActionCommand. Please see my update below.